scripts/image

```diff
@@ -310,32 +310,66 @@ fi
           KERNEL_NAME="KERNEL"
         fi
 
-        UBOOT_SYSTEM=""
-        if [ "$PROJECT" == "imx6" ]; then
-          if [ -n "$SYSTEM" ]; then
-            UBOOT_SYSTEM="$SYSTEM"
+        # default target for image
+        if [ "$PROJECT" = "imx6" ]; then
+          if [ -z "$IMAGES" ]; then
+            UBOOT_IMAGES=$IMAGES_SYSTEMS_NEEDED
           else
-            UBOOT_SYSTEM="cuboxi"
+	    for IMAGE_ASKED in $IMAGES; do
+	      SYSTEM_NEED_IMAGE=false
+	      for IMAGE_SYSTEM_NEEDED in $IMAGES_SYSTEMS_NEEDED; do
+		if [ $IMAGE_SYSTEM_NEEDED = $IMAGE_ASKED ]; then
+                  SYSTEM_NEED_IMAGE=true
+		fi
+              done
+              if [ $SYSTEM_NEED_IMAGE == true ]; then
+                UBOOT_IMAGES+=( $SYSTEM_ASKED )
+              else
+                echo "mkimage: '$SYSTEM_ASKED' is not a valid DEVICE name for build image."
+              fi
+            done
           fi
+        else
+          UBOOT_IMAGES=( "undef" )
```

Comment:
This "undef" is rather ugly.

Also, using the variable name `UBOOT_IMAGES` for what may not be a uboot-based system (ie. RPi, x86 etc.) seems inappropriate. A more generic naming solution would be better, and the undef should be eliminated.